### PR TITLE
PRC-52: Add validation to create contact

### PR DESCRIPTION
**Pull request by james-baker-moj**
Added field length constraints with custom error messages to 1) be helpful to client and 2) stop exposing SQL error messages if the fields are too long.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
+  @Schema(description = "The last name of the contact", example = "Doe", maxLength = 35)
+  @field:Size(max = 35, message = "lastName must be <= 35 characters")
+  @field:NotNull
   val lastName: String,
 
-  @Schema(description = "The first name of the contact", example = "John")
+  @Schema(description = "The first name of the contact", example = "John", maxLength = 35)
+  @field:Size(max = 35, message = "firstName must be <= 35 characters")
+  @field:NotNull
   val firstName: String,
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt:26]: Do these need to be nullable for the jakarta constraints to work as expected? I believe if null, the kotlin error would trump the jakarta validation of `field:notnull` ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt [hunk 2]
```diff
@@ -1,26 +1,42 @@
...
+  @field:DateTimeFormat(pattern = "yyyy-MM-dd")
+  @field:Nullable
   val dateOfBirth: LocalDate? = null,
 
-  @Schema(description = "The id of the user creating the contact", example = "JD000001")
+  @Schema(description = "The id of the user creating the contact", example = "JD000001", maxLength = 100)
+  @field:Size(max = 100, message = "createdBy must be <= 100 characters")
+  @field:NotNull
   val createdBy: String,
 )
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt:42]: Create time - might need a json formatter to limit to hours, minutes, seconds?

**Review by timharrison-moj — DISMISSED** [review]
LGTM. Just a couple of minor, around the edges, comments.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt (hunk 1 above)
Comment (james-baker-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt:26]: Yeah I discovered that these don't work hence the error handling changes but forgot to remove it. It's already described as not null in the kotlin and swagger so a 3rd place is just noise now.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt (hunk 2 above)
Comment (james-baker-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/CreateContactRequest.kt:42]: Create time is currently done server side with LocalDateTime.now(), this is just the user id.

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**